apps/fabric-website/src/pages/NotFoundPage/NotFoundPage.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { Link, DefaultButton, mergeStyles } from 'office-ui-fabric-react';
+import {
+  trackEvent,
+  EventNames,
+  randomEntry,
+  getSiteArea,
+  Page,
+  IPageProps,
+  IPageSectionProps
+} from '@uifabric/example-app-base/lib/index2';
+import { SiteDefinition } from '../../SiteDefinition/index';
+import { topNavHeight } from '../../styles/constants';
+
+const illustrations = [
+  'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/images/error/error1.svg',
+  'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/images/error/error2.svg',
+  'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/images/error/error3.svg'
+];
+
+const rootClass = mergeStyles({
+  backgroundRepeat: 'no-repeat',
+  backgroundSize: '480px',
+  backgroundPosition: '80% 80%',
+  backgroundImage: `url(${randomEntry(illustrations)})`,
+  minHeight: `calc(100vh - ${topNavHeight}px)`
+});
+
+export interface INotFoundPageProps extends IPageProps {}
+
+export class NotFoundPage extends React.Component<INotFoundPageProps, {}> {
+  public componentDidMount(): void {
+    trackEvent(EventNames.PageNotFound, {
+      currentArea: getSiteArea(),
+      currentPage: window.location.hash
+    });
+  }
+
+  public render() {
+    return <Page title="Page not found" className={rootClass} otherSections={this._otherSections()} showSideRail={false} />;
+  }
+
+  private _otherSections = (): IPageSectionProps[] => [
+    {
+      sectionName: 'Sorry, the page you requested cannot be found.',
+      content: (
+        <div>
+          <p>The URL may be misspelled or the page you are looking for is no longer available.</p>
+          <ul>
+            <li>
+              <Link
+                href="#/"
+                onClick={
+                  // tslint:disable-next-line jsx-no-lambda
+                  ev => this._onInternalLinkClick(ev, '#/')
+                }
+              >
+                UI Fabric Home
+              </Link>
+            </li>
+            {this._getAreaLink()}
+          </ul>
+          {this._renderBackButton()}
+        </div>
+      )
+    }
+  ];
+
+  /** Gets the top level page from the current URL and returns a link to it. */
+  private _getAreaLink = (): JSX.Element => {
+    const area = getSiteArea(SiteDefinition.pages);
+    const SiteDefinitionFilter = SiteDefinition.pages.filter(page => page.title === area);
```

Comment:
Please use camelCase variable names. Also, a name like pagesForArea would be more helpful/descriptive here--or just pageForArea if you go ahead and grab the first element out of the array (this will still work on an empty array because in JS, accessing an array index which doesn't exist returns undefined).
```suggestion
    const pageForArea = SiteDefinition.pages.filter(page => page.title === area)[0];
```

Reply:
Whoops, that was a find/replace mistake. Your suggestion is good.